### PR TITLE
DOC: Add cross merge to userguide section

### PR DESCRIPTION
**Pull request by phofl**
Forgot this when adding cross

### CODE REVIEW

File: doc/source/user_guide/merging.rst [hunk 1]
```diff
@@ -707,6 +707,7 @@ either the left or right tables, the values in the joined table will be
     ``right``, ``RIGHT OUTER JOIN``, Use keys from right frame only
     ``outer``, ``FULL OUTER JOIN``, Use union of keys from both frames
     ``inner``, ``INNER JOIN``, Use intersection of keys from both frames
+    ``cross``, ``CROSS JOIN``, Create the karthesian product of rows of both frames
```
Comment (jreback) [inline doc/source/user_guide/merging.rst:710]: isin't this cartesian?

**Comment by jreback** [comment]
ok to backport this

File: doc/source/user_guide/merging.rst (hunk 1 above)
Comment (phofl) [reply doc/source/user_guide/merging.rst:710]: Yes thx

**Comment by phofl** [comment]
Docs look reasonable

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Pull request merged**